h/views/api/flags.py

```diff
@@ -18,19 +19,28 @@
 def create(context, request):
     request.find_service(name="flag").create(request.user, context.annotation)
 
-    _email_group_admin(request, context.annotation)
+    _email_group_moderators(request, context.annotation)
 
     return HTTPNoContent()
 
 
-def _email_group_admin(request, annotation):
+def _email_group_moderators(request, annotation):
     incontext_link = links.incontext_link(request, annotation)
     if incontext_link is None:
         incontext_link = annotation.target_uri
 
-    group = annotation.group
-    if group.creator and group.creator.email:
-        send_params = flag_notification.generate(
-            request, group.creator.email, incontext_link
-        )
-        mailer.send.delay(*send_params)
+    group_members_service = request.find_service(name="group_members")
+
+    memberships = group_members_service.get(
+        annotation.group,
+        roles=[
+            GroupMembershipRoles.OWNER,
+            GroupMembershipRoles.ADMIN,
+            GroupMembershipRoles.MODERATOR,
+        ],
+    )
```

Comment:
This isn't ideal: it hard-codes the logic that owners, admins and moderators are the ones who can moderate groups, which is duplicated [in the permission map](https://github.com/hypothesis/h/blob/5f632e2fcba488d0675ff9bbfc20b90d665524f1/h/security/permission_map.py#L58-L62).

It would be better if this went through the permissions system to get the list of users who have the `Permission.Group.MODERATE` permission in the context of this annotation's group: create a `context` object for the group, then for each member create an `identity` object and call [`identity_permits(identity, context, Permission.Group.MODERATE)`](https://github.com/hypothesis/h/blob/8765297d0f6c1c3df82d8d64087dd2663d77ccb1/h/security/permits.py#L9-L31) But there are two problems with that:

1. We'd have to fetch all the group's members from the DB and iterate over them, doing a permissions check for each individual member to find out which ones to email
2. Creating an identity for a user with `Identity.from_models(user)` [iterates over all the user's memberships](https://github.com/hypothesis/h/blob/8765297d0f6c1c3df82d8d64087dd2663d77ccb1/h/security/identity.py#L65) (of all groups that the user is a member of). So we don't want to iterate over all a group's memberships calling `Identity.from_models(membership.user)`  for each of them

We might be able to fix (2) by having `Identity.from_models()` accept an optional list of memberships to use instead of `user.memberships`. Then the view could fetch the group's memberships once and use them for all permissions checks. Something like this:

```python
context = GroupContext(group)
for membership in group.memberships:
    identity = Identity.from_models(user, memberships=[membership])
    if identity_permits(identity, context, Permission.Group.MODERATE):
        # Send email;
```

This might be worth doing because https://github.com/hypothesis/h/pull/9115 has a similar problem that it might want to solve by calling `Identity.from_models(user)` without having it query for all of `user`'s memberships.